api/specs.proto

```diff
@@ -118,6 +107,19 @@ message TaskSpec {
 	// LogDriver specifies the log driver to use for the task. Any runtime will
 	// direct logs into the specified driver for the duration of the task.
 	Driver log_driver = 6;
+
+	// Networks specifies the list of network attachment
+	// configurations (which specify the network and per-network
+	// aliases) that this task spec is bound to.
+	repeated NetworkAttachmentConfig networks = 7;
+}
+
+// NetworkAttachmentSpec specifies runtime parameters required to attach
+// a container to a network.
+message NetworkAttachmentSpec {
+	// ID spcifies a unique ID of the entity for which this
+	// attachment is for.
+	string id = 1 [(gogoproto.customname) = "ID"];
```

Comment:
After come thought, it is probably clearer to call this `ContainerID`. We should probably also include an attachment id, if that is a thing.